protoarray/src/test/java/tech/pegasys/teku/protoarray/ProtoArrayTest.java

```diff
@@ -469,6 +469,62 @@ void contains_shouldNotContainDescendantsOfInvalidBlock() {
     assertThat(protoArray.contains(block2a)).isFalse();
   }
 
+  @Test
+  void findMergeTransitionBlock_emptyWhenHeadNotKnown() {
+    assertThat(protoArray.findMergeTransitionBlock(block1a)).isEmpty();
+  }
+
+  @Test
+  void findMergeTransitionBlock_emptyWhenTransitionNotYetHappened() {
+    addOptimisticBlock(1, block1a, GENESIS_CHECKPOINT.getRoot(), Bytes32.ZERO);
+    assertThat(protoArray.findMergeTransitionBlock(block1a)).isEmpty();
+  }
+
+  @Test
+  void findMergeTransitionBlock_emptyWhenTransitionBeforeStartOfProtoArray() {
+    addOptimisticBlock(1, block1a, GENESIS_CHECKPOINT.getRoot());
+    addOptimisticBlock(2, block2a, block1a);
+    addOptimisticBlock(3, block3a, block2a);
+    assertThat(protoArray.findMergeTransitionBlock(block3a)).isEmpty();
+  }
+
+  @Test
+  void findMergeTransitionBlock_emptyWhenAfterSpecifiedHead() {
+    addOptimisticBlock(1, block1a, GENESIS_CHECKPOINT.getRoot(), Bytes32.ZERO);
+    addOptimisticBlock(2, block2a, block1a, Bytes32.ZERO);
+    addOptimisticBlock(3, block3a, block2a, Bytes32.ZERO);
+    addOptimisticBlock(3, block4a, block3a);
```

Comment:
shouldn't be slot 4?

Reply:
It doesn't actually matter to protoarray but yes that would make more sense.